### PR TITLE
Add comments for exported functions and interfaces in roosterjs-editor-api/lib/format/ (part1)

### PR DESCRIPTION
**Pull request by redmondbowang**

### CODE REVIEW

File: packages/roosterjs-editor-api/lib/format/removeLink.ts [hunk 1]
```diff
@@ -3,7 +3,11 @@ import queryNodesWithSelection from '../cursor/queryNodesWithSelection';
 import { normalizeEditorPoint, unwrap } from 'roosterjs-editor-dom';
 import { Editor } from 'roosterjs-editor-core';
 
-export default function removeLink(editor: Editor): void {
+/**
+ * Remove link at selection, if no links at selection, do nothing
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/removeLink.ts]: explain what will happen when selection contains multiple links.
explain what will happen if only part of a link is selected. e.g. <a>THIS IS {Sel Start} A LINK </a> {Sel End}

File: packages/roosterjs-editor-api/lib/format/insertTable.ts [hunk 2]
```diff
@@ -12,6 +21,14 @@ export interface TableFormat {
     borderCollapse?: string;
 }
 
+/**
+ * Insert table into editor at current selection
+ * @param editor The editor instance
+ * @param columns Number of columns in table, it also controls the default table cell width:
+ * if columns <= 4, width = 120px; if columns <= 6, width = 100px; else width = 70px
+ * @param rows Number of rows in table
+ * @param format (Optional) The table format
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/insertTable.ts]: If this is null, what's the default format

File: packages/roosterjs-editor-api/lib/format/insertTable.ts [hunk 3]
```diff
@@ -3,6 +3,15 @@ import execFormatWithUndo from './execFormatWithUndo';
 
 const ZERO_WIDTH_SPACE = '&#8203;';
 
+/**
+ * The table format
+ * @param cellSpacing (Optional) The cellSpacing style for the HTML table element
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/insertTable.ts]: Don't add comment for interface member here, but add comment to each member, e.g.

/**
 The cellPadding style for the HTML table element
*/
cellPadding?: string;
...

File: packages/roosterjs-editor-api/lib/format/insertImage.ts [hunk 4]
```diff
@@ -1,5 +1,11 @@
 import { Editor } from 'roosterjs-editor-core';
 
+/**
+ * Insert an image to editor at current selection
+ * @param editor The editor instance
+ * @param imageFile The image file. There are 3 ways to obtain the file object:
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/insertImage.ts]: There are at least 3 ways to ...

File: packages/roosterjs-editor-api/lib/format/getFormatState.ts [hunk 5]
```diff
@@ -45,6 +45,10 @@ function queryCommandState(editor: Editor, command: string): boolean {
 /**
  * Get format state at cursor
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/getFormatState.ts]: Please add a brief introduction what is format state.

File: packages/roosterjs-editor-api/lib/format/execFormatWithUndo.ts [hunk 6]
```diff
@@ -10,12 +10,19 @@ import { EditorPoint, NodeType } from 'roosterjs-editor-types';
  */
 export type Formatter = (startPoint: EditorPoint, endPoint: EditorPoint) => Node | void | any;
 
-// Exec format with undo
+/**
+ * Execute format with undo
+ * @param editor The editor instance
+ * @param formatter The callback format function we want to perform, it also creates a fallback node for selection
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/execFormatWithUndo.ts]: Explain what is fallback node

File: packages/roosterjs-editor-api/lib/format/execFormatWithUndo.ts [hunk 7]
```diff
@@ -10,12 +10,19 @@ import { EditorPoint, NodeType } from 'roosterjs-editor-types';
  */
 export type Formatter = (startPoint: EditorPoint, endPoint: EditorPoint) => Node | void | any;
 
-// Exec format with undo
+/**
+ * Execute format with undo
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/execFormatWithUndo.ts]: Need to mention, it tries to add undo snapshot at begin and end of the function. And duplicated snapshot will be only added once.

File: packages/roosterjs-editor-api/lib/format/clearFormat.ts [hunk 8]
```diff
@@ -6,7 +6,13 @@ import setTextColor from './setTextColor';
 import { Editor } from 'roosterjs-editor-core';
 import queryNodesWithSelection from '../cursor/queryNodesWithSelection';
 
-export default function clearFormat(editor: Editor): void {
+/**
+ * Clear the format in current selection, after cleaning, the format will be
+ * changed to default format. The format that get cleaned include B/I/U/font name/
+ * font size/text color/background color/left/right/center/superscript/subscript
+ * @param editor The editor instance
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/clearFormat.ts]: what is left/right/center?

Comment (redmondbowang) [reply packages/roosterjs-editor-api/lib/format/clearFormat.ts]: Changed to align left/right/center

File: packages/roosterjs-editor-api/lib/format/execFormatWithUndo.ts (hunk 7 above)
Comment (redmondbowang) [reply packages/roosterjs-editor-api/lib/format/execFormatWithUndo.ts]: Added

File: packages/roosterjs-editor-api/lib/format/execFormatWithUndo.ts (hunk 6 above)
Comment (redmondbowang) [reply packages/roosterjs-editor-api/lib/format/execFormatWithUndo.ts]: Added

File: packages/roosterjs-editor-api/lib/format/getFormatState.ts (hunk 5 above)
Comment (redmondbowang) [reply packages/roosterjs-editor-api/lib/format/getFormatState.ts]: Added

File: packages/roosterjs-editor-api/lib/format/insertImage.ts (hunk 4 above)
Comment (redmondbowang) [reply packages/roosterjs-editor-api/lib/format/insertImage.ts]: Changed.

File: packages/roosterjs-editor-api/lib/format/insertTable.ts (hunk 3 above)
Comment (redmondbowang) [reply packages/roosterjs-editor-api/lib/format/insertTable.ts]: Changed.

File: packages/roosterjs-editor-api/lib/format/insertTable.ts (hunk 2 above)
Comment (redmondbowang) [reply packages/roosterjs-editor-api/lib/format/insertTable.ts]: Added null case

File: packages/roosterjs-editor-api/lib/format/removeLink.ts (hunk 1 above)
Comment (redmondbowang) [reply packages/roosterjs-editor-api/lib/format/removeLink.ts]: Added.

**Pull request merged**